Assets/Scripts/SceneLoader.cs

```diff
@@ -1,113 +1,126 @@
+
+    // Stores max number of saves required per star per level
+    private List<List<int>> savesRequired = new List<List<int>>();
+
+    // Read information about max number of saves required per star per level
+    private void ParseStarInfo()
     {
+        string text = File.ReadAllText("./Assets/Resources/starinfo.txt");
+        string[] lines = text.Split('\n');
+
+        savesRequired = new List<List<int>>();
+        foreach (string line in lines)
+        {
+            if (line == "") break;
+            string[] info = line.Split(' '); // level number, 1 star, 2 star, 3 star
+            List<int> saves = new List<int> (new int[] { int.Parse(info[1]), int.Parse(info[2]), int.Parse(info[3]) } );
+            savesRequired.Add(saves);
+        }
+    }
 
-        if (PlayerPrefs.HasKey("starprogress"))
+    // Retrieve data about player's smallest ever no. of saves per level
+    // -1 indicates the level is locked, 0 indicates the level is unlocked but not attempted
+    private void GetPlayerProgress()
+    {
+        if (PlayerPrefs.HasKey("playerMinSaves"))
         {
-            starprogress = PlayerPrefs.GetString("starprogress");
+            playerMinSaves = PlayerPrefs.GetString("playerMinSaves");
         }
         else
         {
-            starprogress = "0;-1;-1;-1;-1;-1;-1;-1;-1;-1";
-            PlayerPrefs.SetString("starprogress", starprogress);
+            PlayerPrefs.SetString("playerMinSaves", playerMinSaves);
         }
+    }
 
-        string[] starSplit = starprogress.Split(';');
+    // Populate menu stars according to playerMinSaves and savesRequired
+    private void PopulateStars()
+    {
+        int[] minSaves = Array.ConvertAll(playerMinSaves.Split(';'), s=> int.Parse(s));
         int currentLevel = 1;
-        foreach (string star in starSplit)
+        foreach (int saves in minSaves)
         {
-            GameObject level = Instantiate(LevelSelector, LevelContent.transform.position, Quaternion.identity);
+            List<int> savesNeeded = savesRequired[currentLevel-1];
 
+            GameObject level = Instantiate(LevelSelector, LevelContent.transform.position, Quaternion.identity);
             level.transform.parent = LevelContent.transform;
             level.transform.localScale = new Vector3(1, 1, 1);
+            level.name = currentLevel.ToString() + "-level" + currentLevel.ToString();
 
-            if (star == "-1")
+            if (saves == -1) // locked level
             {
-                level.name = currentLevel.ToString() + "-level" + currentLevel.ToString();
                 level.transform.GetChild(0).gameObject.SetActive(false);
                 level.transform.GetChild(1).gameObject.SetActive(true);
                 level.transform.GetChild(2).gameObject.SetActive(false);
                 level.transform.GetChild(3).gameObject.SetActive(false);
                 level.transform.GetChild(4).gameObject.SetActive(false);
```

Comment:
Refactor by moving into a class LevelIcon. Expose a way to change the icons (5 states: 0 star, 1 star, 2 star, 3 star, locked). LevelIcon.showIcon(SOME_STATE_NUMBER).